Frontend/src/app/hospital-map/buildings/buildings.component.ts

```diff
@@ -0,0 +1,88 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import * as d3 from 'd3';
+import { BuildingsService } from './buildings.service';
+import { SurroundingObjectsService } from './surrounding-objects.service';
+
+@Component({
+  selector: 'app-buildings',
+  templateUrl: './buildings.component.html',
+  styleUrls: ['./buildings.component.css'],
+  providers: [BuildingsService, SurroundingObjectsService]
+})
+export class BuildingsComponent implements OnInit {
+  mainBuilding = [
```

Comment:
Add a polygon above the main building and let the main building be just one polygon instead of a map.